server/data/hmppsRestClient.ts

```diff
@@ -38,36 +40,41 @@ interface StreamRequest {
   errorLogger?: (e: UnsanitisedError) => void
 }
 
-export default class RestClient {
-  agent: Agent
+interface SignedWithMethod {
+  token?: string
+  username?: string
+}
```

Comment:
I like this, and how it switches all calls between either the user token, if present, or generates a system token with or without the username.